src/platform/plugins/shared/share/public/components/tabs/embed/embed_content.tsx

```diff
@@ -267,22 +318,30 @@ export const EmbedContent = ({
         <EuiSpacer />
       </EuiForm>
       <EuiFlexGroup justifyContent="flexEnd" responsive={false}>
+        <React.Fragment>
+          {showPublicUrlSwitch ? renderPublicUrlOptionsSwitch() : null}
+        </React.Fragment>
         <EuiFlexItem grow={false}>
-          <EuiCopy textToCopy={url}>
-            {(copy) => (
-              <EuiButton
-                data-test-subj="copyEmbedUrlButton"
-                onClick={copy}
-                data-share-url={url}
-                fill
-              >
-                <FormattedMessage
-                  id="share.link.copyEmbedCodeButton"
-                  defaultMessage="Copy embed code"
-                />
-              </EuiButton>
-            )}
-          </EuiCopy>
+          <EuiToolTip
+            content={
```

Comment:
Wonder if we can simplify using EUI helper? https://eui.elastic.co/#/utilities/copy#copy-to-clipboard-function

Reply:
Hey @Dosant do you mean the `EuiCopy` component because we use the `copyToClipboard` helper already, the `copyToClipboard` function is used here because in the case where the url that would be copied is a short one we'd have to value to pass to the `EuiCopy` component at the time it's clicked doesn't exist, hence why we opt to use the `copyToClipboard` helper.

Reply:
ok, got it, thanks